client/java/src/main/java/io/openlineage/client/utils/OpenLineageEnvParser.java

```diff
@@ -0,0 +1,154 @@
+        .forEach(parsedJson -> merge(root, parsedJson));
+    String jsonString = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(root);
+    return jsonString;
+  }
+
+  /**
+   * Parses a single environment variable key and value into a nested JSON structure.
+   *
+   * @param envKey The environment variable key.
+   * @param envValue The environment variable value.
+   * @return A nested JSON structure based on the key.
+   */
+  private static ObjectNode parseEnvToNestedJson(String envKey, String envValue) throws Exception {
+    // Remove the 'OPENLINEAGE__' prefix
+    String strippedKey = envKey.substring(PREFIX.length());
+
+    // Split the remaining key on '__' into a List<String>
+    List<String> keyParts =
+        Stream.of(strippedKey.split("__"))
+            .map(String::toLowerCase) // Lowercase the keys if needed
+            .map(OpenLineageEnvParser::convertSnakeCaseToCamelCase)
+            .collect(Collectors.toList());
+
+    // Create the final value node. If it's valid JSON, parse it; otherwise, treat
+    // it as a string.
+    JsonNode valueNode;
+    try {
+      valueNode = objectMapper.readTree(envValue);
+    } catch (Exception e) {
+      // If it's not valid JSON, treat it as a plain string
+      valueNode = objectMapper.convertValue(envValue, JsonNode.class);
+    }
+
+    // Create nested structure based on keyParts
+    ObjectNode root = objectMapper.createObjectNode();
+    ObjectNode currentNode = root;
+
+    // Create nested objects for each key part except the last
+    for (int i = 0; i < keyParts.size() - 1; i++) {
+      ObjectNode nextNode = objectMapper.createObjectNode();
+      currentNode.set(keyParts.get(i), nextNode); // Use List's get() method
+      currentNode = nextNode;
+    }
+
+    // Set the final value at the last part of the key
+    currentNode.set(keyParts.get(keyParts.size() - 1), valueNode);
+
+    return root;
+  }
+
+  /**
+   * Converts a string from snake_case to camelCase.
+   *
+   * @param value The input string in snake_case.
+   * @return The input string converted to camelCase.
+   */
+  private static String convertSnakeCaseToCamelCase(String value) {
+    return value.indexOf('_') != -1
+        ? value.substring(0, value.indexOf('_'))
+            + Arrays.stream(value.substring(value.indexOf('_') + 1).split("_"))
+                .map(s -> Character.toUpperCase(s.charAt(0)) + s.substring(1))
+                .collect(Collectors.joining())
+        : value;
+  }
+
+  /**
+   * Merges two ObjectNodes into one.
+   *
+   * @param mainNode The main ObjectNode that will be merged into.
+   * @param updateNode The ObjectNode to merge.
+   */
+  private static void merge(ObjectNode mainNode, ObjectNode updateNode) {
+    updateNode
+        .fields()
+        .forEachRemaining(
+            entry -> {
+              String fieldName = entry.getKey();
+              JsonNode value = entry.getValue();
+
+              if (mainNode.has(fieldName)) {
```

Comment:
What do you think about keeping code (happy path) left + early return 
```java
    if (!mainNode.has(fieldName)) {
      mainNode.set(fieldName, value);
      return;
    }

    // If the field already exists and both are objects, recursively merge
    JsonNode existingValue = mainNode.get(fieldName);
    if (existingValue.isObject() && value.isObject()) {
      merge((ObjectNode) existingValue, (ObjectNode) value);
      return;
    }

    mainNode.set(fieldName, value);
```


I think that way code is simpler to read